Src/VimCore/CommonOperations.fs

```diff
@@ -1586,6 +1586,34 @@ type internal CommonOperations
     member x.RaiseSearchResultMessage searchResult = 
         CommonUtil.RaiseSearchResultMessage _statusUtil searchResult
 
+    /// Record last change or yank start and end positions
+    member x.RecordLastChangeOrYank oldSpan newSpan =

```

Comment:
I think we should consider breaking this into two methods: RecordLastChange and RecordLastYank. The meaning of the input differs significantly based on the scenario. 

One pattern that usually causes me concern is when there are multiple snapshot inputs to a method (`SnapshotSpan`, `SnapshotPoint`, etc ...) where sometimes they are on the same snapshot and other times they aren't. Can lead to cases where code works great in a few scenarios but then falls down where it hits the "other" case. Sometimes it's necessary to have this in an API but I try to avoid it if possible.

Splitting it up would seem to mean that:

- Yank always has one span 
- Delete has two and they're never on the same snapshot.

Reply:
Ha, ha, I tried to get clever! I used to have just one span (the new span) and a bool `isYank`. Then I realized I needed the old span to detect deletes. Then I thought I could delete the bool by using the spans themselves to represent yanks as oldSpan = newSpan.

I agree now that it is too obtuse.